cmd/git.go

```diff
@@ -19,7 +19,19 @@ func ExtractGitInfo() (string, string, string, string, error) {
 		return "", "", "", "", fmt.Errorf("Could not get project Url: %v", err)
 	}
 
-	re := regexp.MustCompile(`(?:[:\/])([^\/]+)\/([^\/]+)\.git$`)
+	/*
+	   This should match following formats:
+	       namespace: namespace, projectName: dummy-test-repo:
+	           https://gitlab.com/namespace/dummy-test-repo.git
+	           git@gitlab.com:namespace/dummy-test-repo.git
+	           ssh://git@gitlab.com/namespace/dummy-test-repo.git
+
+	       namespace: namespace/subnamespace, projectName: dummy-test-repo:
+	           ssh://git@gitlab.com/namespace/subnamespace/dummy-test-repo
+	           https://git@gitlab.com/namespace/subnamespace/dummy-test-repo.git
+	           git@git@gitlab.com:namespace/subnamespace/dummy-test-repo.git
+	*/
+	re := regexp.MustCompile(`(?:https?:\/\/|ssh:\/\/|git@)(?:[^\/:]+)[\/:](.*)\/([^\/]+?)(?:\.git)?$`)
```

Comment:
I found out that on many repositories I often use very different format of url ( I don't set it manually ) so I updated regexp to match urls in comment. I tested this manually and also [here](https://regex101.com/r/UU8IBo/1). The main problem was the assumption that namespace is just one word ( `([^\/]+` ) but namespace is everything in URI till the project name.

Reply:
Yeah, good call, this is an awesome catch. I just merged in another contribution (https://github.com/harrisoncramer/gitlab.nvim/pull/87) that fixes this (as well as adding some other checks + tests) for the regex.

If you resolve the merge conflicts on your branch (with the latest merge) I'll merge this in. Thank you for your patience and for dealing with multiple conflicts!

Reply:
I resolved the conflict here but my regexp is more extended - the previous version was for example missing `ssh://` and also the `.gitf

Reply:
I also added tests.